drivers/soundwire/intel_init.c

```diff
@@ -474,7 +532,7 @@ void sdw_intel_process_wakeen_event(struct sdw_intel_ctx *ctx)
 		if (!(link_mask & BIT(i)))
 			continue;
 
-		intel_master_process_wakeen_event(link->pdev);
+		intel_link_process_wakeen_event(link->ldev);
```

Comment:
I could be wrong, Pierre but it feels like cts->links should point to struct sdw_intel_link_dev and not struct sdw_intel_link_res. That way you wont need this circular pointer dependency between link_dev and link_res.

Reply:
yes exactly what I wanted to change, but kicked to the next morning :-)